### ISSUE TITLE
Start showing some craft and office POIs

### ISSUE THREAD

**Issue by kocio-pl** [gravitystorm/openstreetmap-carto#1697]
I think we should show not just shops and amenities, but also some objects with [craft=*](http://taginfo.openstreetmap.org/keys/craft#values) and [office=*](http://taginfo.openstreetmap.org/keys/office#values) key. Case by case analysis is probably needed to know which we want to show and how - sometimes the same icon and/or color may be used as with already rendered POIs, but when it doesn't match, maybe generic black/brown/violet dot and label would work.

Looking at the most popular POIs from these categories:

* craft (above 1k of uses):
 * carpenter - icon needed
 * shoemaker - brown icon, probably not the same as shop=shoe
 * photographer - probably the same as shop=photo
 * electrician - brown flash icon
 * brewery - icon needed
 * tailor - icon needed, probably the same as shop=tailor
 * plumber - icon needed
 * metal_construction - icon needed
 * confectionery - maybe the same as shop=confectionery, but brown or black?
 * gardener - maybe the same as shop=garden_centre, but brown?

* office (above 5k uses):
 * company - maybe generic black dot from z>=19?
 * government - brown generic dot or [government icon](https://github.com/nebulon42/osmic/blob/master/administration/government-14.svg)
 * estate_agent - icon needed, violet as shops
 * yes - generic black dot (with name)
 * insurance - icon needed
 * administrative - brown generic dot or town hall icon

**Comment by math1985** [comment]
See also #108.

**Comment by pnorman** [comment]
Given we already have too many different icons, we need to be careful about this.

We also need consider where we're going cartographically. For a general purpose map, I would be against rendering office=yes.

**Comment by polarbearing** [comment]
amenity-brown is mostly dedicated towards touristic/cultural features (see also #1624), it should not be expanded to commercial activities as in crafts.

**Comment by kocio-pl** [comment]
@polarbearing This interpretation is too narrow - what about post office, town hall, police station, fire station or even courthouse? Amenity is all about different services, while shop is all about buying goods. That was my conclusion [when thinking about where the bank/atm belong](https://github.com/gravitystorm/openstreetmap-carto/pull/1658#issue-95012587) and now it makes a lot of sense for me.

@pnorman You're right, after leaving safe shops/amenities area we're on *terra incognita* and we have to watch our steps carefully. I am not against rendering office=yes, but I definitely want to take some precautions here (which I didn't mentioned just to keep the list compact):
- I would do it at the lowest possible level (z>=19)
- I would do it after gaining some experience with other craft/office POIs

Cartographically there's not too much things on highest levels to show and office=yes may be as useful for some people as shop=yes.

**Comment by vince-from-nice** [comment]
:+1:  I completely  agree, the possibility to view office POIs would be something very valuable for me.. even at Z=18 !  :)

**Comment by planemad** [comment]
^ especially `office= government` which is of public importance

**Comment by matkoniecz** [comment]
copy from #1966 

for craft=brewery - it is for

- where it is impossible to buy anything (otherwise it would have pub or bar icon or at least would be tagged as shop)
- it is impossible to visit them for tour or something similar (it would be tourism=museum/attraction) 
- are not big enough to draw even small landuse 
- are even not filling entire building (building names are rendered)

I am unsure whatever rendering name would be valuable (alcohol-related icon would be clearly misleading as it is not bar/pub/shop so it is unacceptable).

**Comment by matkoniecz** [comment]
In general - I strongly oppose new icon, labels are likely to be useful for offices. I am unsure about labels for craft.

**Comment by kocio-pl** [comment]
As I look here now, designing right icons may be hard, so I would start with showing at least labels with dots for those keys.

**Comment by SomeoneElseOSM** [comment]
I'd agree that "labels with dots" for offices makes sense too.  Over at https://github.com/SomeoneElseOSM/SomeoneElse-style/blob/master/style.lua I mapped a long list of nonspecific shop, office, healthcare and leisure items to purple, black, pink and green dots.  It probably wouldn't be feasible or desirable to have as along a list in osm-carto, but I think the approach makes sense.

**Comment by kocio-pl** [comment]
I think the difference between amenity, shop, craft and office is somewhat blurred - especially we don't know what "amenity" really is (and that's probably just a legacy general type for POIs). But we also don't know what is "craft" - it can be "amenity" or "shop" if we think about tailor for example, since it can be a workshop alone (that is strictly craft), but also the place you can bring your coat for repair ("shop" as in hairdresser currently, but if you think that "shop is all about goods and amenity for services" that would be "amenity" for sure). 

The result is that I don't want to be strict about icons for those types. I would put:
- (brown) paragraph sign for a lawyer, no matter if it's tagged as an "office" or "amenity"
- (brown) needle/button for a tailor, no matter if it's "craft" or "amenity" 
- (violet?) house icon for real estate agency, no matter if it's "office" or maybe "shop"

because for simple map user these are typical, easily recognizable POIs and it's just our internal problem that we don't have clear ontology they could fit in.

So I would propose to start rendering craft and office as a dot+label, but try to make icons case by case if it's possible, not looking too much into the tagging schemes.

**Comment by liondog** [comment]
Whats the current state? Any further discussions? Carto currently shows house numbers instead of names. This leads to repeated house numbers in case of multiple offices. In my eyes a rather bizarre and puzzling rendering result. Wouldn't be a simple dot + name a much better solution?

**Comment by kocio-pl** [comment]
We're discussing implementation details now: https://github.com/gravitystorm/openstreetmap-carto/issues/108#issuecomment-328904162.

**Comment by Tomasz-W** [comment]
I think we should discuss in which cases #3061 is enough, and in which cases we still need an separate icon.

**Comment by Tomasz-W** [comment]
Icons proposals for some craft=* tags:

* craft=brewery
![craft brewery](https://user-images.githubusercontent.com/25656654/36198561-35702ef0-1177-11e8-88be-1561e324ed22.png)
* craft=carpenter
![craft carpenter](https://user-images.githubusercontent.com/25656654/36198562-3594b52c-1177-11e8-9d63-9926b350a5c1.png)
* craft=electrician (-> I'm not sure what is this tag for - a repair point, where you can bring broken device, or kind of office for workers which are typically working outside; if it's a second option, I think a dot would be enough here)
![craft electrician cable](https://user-images.githubusercontent.com/25656654/36198563-35bb10e6-1177-11e8-9121-a101ab9406a7.png)   (cable)
![craft electrician](https://user-images.githubusercontent.com/25656654/36198564-35db16e8-1177-11e8-884b-049f8991c42e.png) (flash)
* craft=plumber
![craft plumber tap](https://user-images.githubusercontent.com/25656654/36198566-362621d8-1177-11e8-9142-0cdb54a8c259.png) (tap)
![craft plumber tools](https://user-images.githubusercontent.com/25656654/36198568-36503aea-1177-11e8-867a-699185cdfed0.png) (tools)
* craft=shoemaker
![craft shoemaker](https://user-images.githubusercontent.com/25656654/36198569-3669940e-1177-11e8-92d3-feba262a3de8.png)
* craft=tailor
![craft tailor](https://user-images.githubusercontent.com/25656654/36198570-3685c534-1177-11e8-9547-ffa0dfc437a6.png)
* craft=metal_construction (-> same thing as in craft=electrician)
* craft=winery (it wasn't mentioned here, but it has 4,4k uses now, so I think we should discuss a separate icon)
* office=* rendering will be resolved soon by https://github.com/gravitystorm/openstreetmap-carto/pull/3061 (I don't think we need separate icons for estate agents and insurance)

**Comment by dieterdreist** [comment]
2018-02-14 11:26 GMT+01:00 Tomasz Wójcik <notifications@github.com>:

> Icons proposals for some craft=* tags:
>

very nice, I like almost all of them,
for the electrician I prefer the cable, the flash could be used for power
stations and substations.
plumber: tap
shoemaker is the only one I find hard to read.

**Comment by Tomasz-W** [comment]
> for the electrician I prefer the cable, the flash could be used for power stations and substations.

I though the same thing, so I did a review od a power=* key. There are only 2 tags which we should consider for flash icon:
* power=plant -> it's rendered as industrial area, so there will be problem with labeling on size (there are many tags with this problem, but discussion about it stuck)
* power=substation -> I've chcecked my neighbourhood and some other places, and I think there are too many substations for rendering them (remember that it's not a common-needed feature)

> shoemaker is the only one I find hard to read.

I'll work on that.

**Comment by andrzej-r** [comment]
There are in fact quite a few office tag values, some of which could benefit from having icons. I can't check it right now but government, lawyer, IT sound like potential targets. 

I would indeed prefer to have #3061 merged first and work on icons later, but that's mostly due to my interest in fixing the mapping feedback issue, not because I don't like icons.

**Comment by Tomasz-W** [comment]
* craft=shoemaker fixed a little bit:
![craft shoemaker](https://user-images.githubusercontent.com/25656654/36437769-3dc2bcda-1668-11e8-8a0e-d549657731f4.png)
* craft=winery icon proposal
![craft winery winogrona](https://user-images.githubusercontent.com/25656654/36437994-dcc7d0b8-1668-11e8-8930-ce55c3e8416b.png) (grapes)
![craft winery beczka](https://user-images.githubusercontent.com/25656654/36437723-16a4452e-1668-11e8-80e8-b1c13add9d88.png) (barrel)

**Comment by kocio-pl** [comment]
Quick remarks:
- craft=winery - I would cut the top of grape cluster, so it will have more triangle shape and maybe bigger leaf (currently it's closer to ananas), but it's still more clear to me than a barrel
- craft=tailor - perfect for me!
- craft=plumber - 1) is better for me
- craft=carpenter - OK
- craft=shoemaker - OK
- craft=brewery - I can't recognize it

**Comment by geowas-github** [comment]
Some remarks: 

- carpenter I don't recognize. I think the saw alone could be do the job,
- winery with grapes is good (however the grapes could be improved maybe... - but they are also ok as they are)
- for the electrician: maybe you can combine the cable/plug or maybe a light bulb with a tool (skrewdriver?). I always like the combination of tool + material/produce/thing being worked on.
- for the plumber I support icon 1.

Shoemaker, tailor and brewery I like.

**Comment by kocio-pl** [comment]
- craft=winery v2 - perfect for me! :+1: 
- craft=brewery - even when I know what it is, I can't recognize it directly; maybe similar design as craft=winery (beer mug and hop)?

**Comment by daganzdaanda** [comment]
I'd take vessel + hop before mug + hop.
plumber - tap
winery v2 - great!
shoemaker - fixed version is good (but maybe the shoe shape could be simplified by merging the soles to reduce the anti-alias blur?)
All the other icons are very good!

Thanks for all the great work on these icons! 
Of course the small details can't all fit with the pixel grid, and so the icons won't be as crisp as many of the bold new SVGs. But it's amazing how well the details still are readable. We should do some test runs on real data to see how they look "in the wild".

**Comment by kocio-pl** [comment]
vessel+hop version mostly works for me too. Maybe mug+hop version would be better if you use this icon (the one you used does look different):
https://github.com/gravitystorm/openstreetmap-carto/blob/master/symbols/biergarten.svg

@daganzdaanda Would you like to make a test PR and some renderings? I'm surprised that you have never been [a code contributor](https://github.com/gravitystorm/openstreetmap-carto/graphs/contributors) to osm-carto, despite the long history of being active here. :smile:

**Comment by daganzdaanda** [comment]
> I'm surprised that you have never been a code contributor 

Yeah, I know ;)  The thing is that I already feel I'm spending too much time on OSM stuff... but thank you guys for starting that Docker thing, maybe sometime I can try that.

**Comment by kocio-pl** [comment]
OK!

We need more active people in this style in general:
- icon (and other cartographic features) designers, 
- coders, 
- community/discussion moderators (to reach some conclusions), 
- simple rendering testers, 
- code reviewers and cleaners (SQL+CartoCSS),
- PR mergers/release managers (only team members can do that)... 

Any activity makes us stronger, but currently we have enough bug reports, nice icons and quite conclusive discussions, I can merge the PRs and try to make monthly releases, so the weakest link is preparing code for PRs at the moment. 

If anyone wants to try, I can help to start it. In my opinion working Docker containers are really important step toward healthy project, yet we need some people to play with it. :smile:

**Comment by Tomasz-W** [comment]
Gist links:

* craft=brewery: https://gist.github.com/Tomasz-W/4c4febd2f19dbfd8ec201100b54946ea
* craft=carpenter: https://gist.github.com/Tomasz-W/fd54bb942d48b0d17931484401a7909b
* craft=electrician (both icons - cable and flash): https://gist.github.com/Tomasz-W/17e626924b80a3a1e009e80d07cbceab
* craft=plumber: https://gist.github.com/Tomasz-W/b2a18f09b17c6e6fcbe1046a13e39f65
* craft=shoemaker: https://gist.github.com/Tomasz-W/0d723d19388ff4c0502d3a18d5750c88
* craft=tailor: https://gist.github.com/Tomasz-W/7227fe37f9e4a7e945a977023af84b25
* craft=winery: https://gist.github.com/Tomasz-W/48e9cfadc53b5cfe99bb34fec08da5f7

**Comment by lakedistrictOSM** [comment]
All of @Tomasz-W 's icons above look good, the only one that might be too complex and less recognisable is the brewery icon. 

Maybe a conical flask and beer bottle? ![brewdraft1png](https://user-images.githubusercontent.com/20193716/37213398-d2a9f36a-23a9-11e8-8a7e-8c0dcc07d289.png) ![brewdraft2png](https://user-images.githubusercontent.com/20193716/37213621-7a687afe-23aa-11e8-9cb6-e6893ce959c1.png)
Or just a barrel? http://g0cqw3d93gz443uvqwvpy0l8.wpengine.netdna-cdn.com/wp-content/themes/bbwnew/img/icons/brewery.svg or https://d30y9cdsu7xlg0.cloudfront.net/png/9236-200.png ?

**Comment by james2432** [comment]
Basic query for craft added(tailor) can add others once approved

**Comment by Tomasz-W** [comment]
As I understood, all from https://github.com/gravitystorm/openstreetmap-carto/issues/1697#issuecomment-367432772 were accepted. @kocio-pl ?

**Comment by james2432** [comment]
it's the only one that is mentonning colour..

**Comment by james2432** [comment]
Are we adding:
 metal_construction
gardener
confectionery?
If so what icons are needed?
Is brown the colour we are going with because it's a service?

**Comment by lakedistrictOSM** [comment]
Are we having a generic dot/symbol for other craft values?

**Comment by Tomasz-W** [comment]
We haven't discuss it yet, but for me there are only 3 colours to consider: 
* amenity-**brown**
* man_made-**grey**
* **orange** (the only one colour don't used now and didn't considered for other features)

The rest (basing on current state and pending discussions):
**violet** is for shops, **blue** for offices, **black** for administrative offices https://github.com/gravitystorm/openstreetmap-carto/pull/3061, **green** for leisure elements and **red** would be propably a healthcare color someday https://github.com/gravitystorm/openstreetmap-carto/issues/2938

**Comment by polarbearing** [comment]
amenity-brown is heavily loaded already, 
man-made-grey would suit.

**Comment by kocio-pl** [comment]
I was thinking about using kind of orange for eating places once, see https://github.com/gravitystorm/openstreetmap-carto/pull/2824#issuecomment-328521710.

**Comment by kocio-pl** [comment]
It might have sense, but I have to see how does it look like.

**Comment by lakedistrictOSM** [comment]
I think we should avoid grey dots where possible because they might look like bollards.

I'd suggest a generic "craft" icon of a spanner and a hammer similar to the one iD uses. Food and drink "crafts" like confectionery can have their own icons. If we use the current confectionery icon (cake) for craft=confectionery then we should design a new icon for shop=confectionery (probably a sweet like the .fr map or one of the ones suggested in https://github.com/gravitystorm/openstreetmap-carto/issues/1534#issuecomment-100043472)

**Comment by Tomasz-W** [comment]
> I think we should avoid grey dots where possible because they might look like bollards.

They don't because craft=* node is usually on building area, so it would be propably intuitive that is not a bollard there.

> I'd suggest a generic "craft" icon

As I know, we don't use generic icons for whole keys in osm-carto

>  If we use the current confectionery icon (cake) for craft=confectionery then we should design a new icon for shop=confectionery

There are examples of the same icon used in different colour for different features (eg. green picnic sites and brown picnic tables or brown art centres and violet art shops)

**Comment by james2432** [comment]
so is it man-made-grey or gastronomy-orange?

**Comment by kocio-pl** [comment]
It's testing time...

**Comment by james2432** [comment]
added screenshots in pull request for all three colours

**Comment by kocio-pl** [comment]
It's shown here: https://github.com/gravitystorm/openstreetmap-carto/pull/3126#issuecomment-387588250.

**Comment by james2432** [comment]
@pnorman doesn't want strokes in svgs

**Comment by kocio-pl** [comment]
There are probably no strokes in our icons and these are only settings for strokes, which are placed by Inkscape automatically.

**Comment by Tomasz-W** [comment]
Any ideas for `craft=metal_construction` icon?

**Comment by james2432** [comment]
@Tomasz-W a couple of I beams stacked on top of eachother?

**Comment by Tomasz-W** [comment]
`craft=metal_constriction` icon proposals:
* ![craft metal_construction](https://user-images.githubusercontent.com/25656654/41481703-40cd7a1c-70d3-11e8-9f39-a4391c3edac4.png) (screws)
* ![craft metal_construction3](https://user-images.githubusercontent.com/25656654/41481706-411c899a-70d3-11e8-874e-4f385add20eb.png) (3 i-beams)
* ![craft metal_construction2](https://user-images.githubusercontent.com/25656654/41481705-40f567fc-70d3-11e8-8648-d1f628a669c5.png) (4 i-beams)

I vote for 'screws'.

**Comment by Tomasz-W** [comment]
Do we have a consensus on a proper colour for craft=* key? Brown seems to be the best, but on the other hand it's heavy loaded with other map features... Another thing to discuss is to add generic dot for craft=* tags without an icon.

**Comment by james2432** [comment]
screws look good to me they look thick enough to be lagg bolts

**Comment by Tomasz-W** [comment]
@james2432 Gist link: https://gist.github.com/Tomasz-W/b83cecde97e3d0351ba3c198b4836cb7

**Comment by Adamant36** [comment]
@Tomasz-W Is there a way to piggy back by off of James2432's craft PR or something to get this done? Or are we stuck waiting until he gets a new laptop? I think the craft icons are an important addition to the map and I'd be willing to do a new PR for them or finish James2432s work after I do a few other things if its possible.

**Comment by Tomasz-W** [comment]
@Adamant36 For more technical questions ask @kocio-pl , I'm an icon designer, not a coder, so unfortunately I can't help you.
@kocio-pl ?

**Comment by kocio-pl** [comment]
Nice to hear you want to do something with that!

Please talk with @james2432 for a start. It's always possible to just create your own PR, because they are independent and may be even competing, as long as they're not merged. But it's better to start talking and do a team work if possible.

**Comment by james2432** [comment]
Well dont copy what i've done exactly because as Paul has stated, the way I did it needs a database reimport, I need to replace craft(and remove the new column in a few files) and replace it with tags->'craft'

**Comment by kocio-pl** [comment]
This is technical detail - it would be easily visible when testing.

What I try to reach now is some simple planning: making sure who wants to do what and when.

**Comment by Tomasz-W** [comment]
Fixed, pixel aligned icons here: https://gist.github.com/Tomasz-W/4c4febd2f19dbfd8ec201100b54946ea

**Comment by Adamant36** [comment]
@jeisenbe, I might be interested in reviving the craft PR if there's a will for it. What do you think?

**Comment by jeisenbe** [comment]
I would be in favor of a simple dot + text rendering similar to office= at
z18 or z19, to start.

Specific icons for certain types could be discussed later.

**Comment by Adamant36** [comment]
That would work. It would definitely make things easier to implement. Any opinion on the color? It looked like both brown and purple were explored in the PR but there was never a decision on which color was best from what I remember.

**Comment by jeisenbe** [comment]
I think they are similar enough to offices (as places of business which are not shops) that we can try using the same color for both `craft=` and `office=`, to avoid having to create another new icon color. There were at least 3 comments in favor of this before: https://github.com/gravitystorm/openstreetmap-carto/pull/3126#issuecomment-419638860,  https://github.com/gravitystorm/openstreetmap-carto/pull/3126#issuecomment-419727349 and https://github.com/gravitystorm/openstreetmap-carto/pull/3126#issuecomment-420213083

(You could also try one of the new colors mentioned in #3641, for example `#12a98d` Lch(62,43,174) - somewhat similar to office, or `#c84a74` Lch(50,54,3) - similar to shop, but this may not be necessary)

**Comment by geowas-github** [comment]
I think a different colour would be better. To me "craft" means manual (not industrial) work (e.g. painter, goldsmith, etc.) in opposite to brain work or administrational work performed in an office. Also people often can drop in here, to discuss the colour of paint, the design of a golden ring, to pick up a produce, etc. So it shares some features with shop as well.
When adding a POI, I always question myself, is this POI _mainly_ about selling (-> shop), administration (-> office) or manual work (-> craft).

**Comment by dieterdreist** [comment]
sent from a phone

> On 3. Sep 2019, at 08:36, geowas-github <notifications@github.com> wrote:
> 
> When adding a POI, I always question myself, is this POI mainly about selling (-> shop), administration (-> office) or manual work (-> craft).


this system is not completely consistent with the actual data, if you look at the top 10 shop values, 3 are about services and one is unclear (yes): hairdresser, beauty and car_repair

Craft can also be combined with shops, e.g. craft=confectionary or craft=optician

**Comment by geowas-github** [comment]
> On 3. Sep 2019, at 08:36, geowas-github ***@***.***> wrote: When adding a POI, I always question myself, is this POI mainly about selling (-> shop), administration (-> office) or manual work (-> craft).
> this system is not completely consistent with the actual data, if you look at the top 10 shop values, 3 are about services and one is unclear (yes): hairdresser, beauty and car_repair Craft can also be combined with shops, e.g. craft=confectionary or craft=optician

Right.

**Comment by polarbearing** [comment]
I'd second jeisenbe and use office colour. All involve some labour.

**Comment by andrzej-r** [comment]
I don't mind mind using office blue either. 

If this was to be more fine grained, I would prefer using either office or shop colours depending on availability of walk-in commercial services.

**Comment by kocio-pl** [comment]
This is similar problem to amenity/shop issues. It is not clear if we should blindly follow keys (shop -> violet, amenity -> brown, office -> blue) or if we should use some internal rules to make the difference. Blind key matching gives more of a tag feeling, while internal rules allows more real-object related feeling. For amenities we go with internal rules for example, for offices we go with blind match currently.

**Comment by kocio-pl** [comment]
We could use some kind of pink also, because it's free now after healthcare objects are rendered in red.

**Comment by IgorEliezer** [comment]
Please add `craft=electronics_repair`, which renders only the housenumber:
* Uses: 3 676
* TagInfo: https://taginfo.openstreetmap.org/tags/?key=craft&value=electronics_repair
* Example: https://www.openstreetmap.org/node/7123561238

**Comment by jeisenbe** [comment]
The key `craft=` is not yet imported as a polygon when mapped as a closed way, so to render these features we need to request the rendering servers reload the database. There are several other changes already approved, so this could happen soon.

Someone will need to submit a PR adding craft to the list in openstreetmap-carto.lua - you can see #3786 for an example of how to do this. 

It's best to first fetch and checkout the `schema_changes` branch, which is also the branch to which the PR needs to be merged.

**Comment by aawiseman** [comment]
Hey all, I just noticed a number of breweries I added to OSM weren't rendering and found this discussion. Personally I like @Tomasz-W 's suggestions from here https://gist.github.com/Tomasz-W/4c4febd2f19dbfd8ec201100b54946ea -- or maybe just a barrel for the brewery? These are often pretty large and popular venues, I think it would be worth showing on the main map. 

I also add `bar=yes` where appropriate, if there's some objections to adding breweries in general, I wonder if that could influence rendering? Maybe if it's `craft=brewery` and `bar=yes` in conjunction then it could show a regular bar icon?

**Comment by jeisenbe** [comment]
@aawiseman - PR #4015 will need to be merged to and then we will have to organize a reload of the rendering database on the openstreetmap.org servers which render this style (and any other database users who update the style will also then need to reload their rendering database). 

I hope this may be done in the next couple of months?

After that it will be possible to consider rendering features such as craft=brewery, if you are interested in submitting a PR to add the rendering.

**Comment by dieterdreist** [comment]
sent from a phone

> On 30. Jan 2020, at 01:21, Andrew Wiseman <notifications@github.com> wrote:
> 
> Maybe if it's craft=brewery and bar=yes in conjunction then it could show a regular bar icon?


there’s also this 
https://taginfo.openstreetmap.org/keys/microbrewery

**Comment by jidanni** [comment]
All I know is e.g., in iD the user types sawmi... and iD offers him sawmill, which is what he wants. He finishes his edit only to be informed by his friends that they never found the sawmill and couldn't call him to ask the location because there is no cell service way out there etc.

**Comment by HolgerJeromin** [comment]
> All I know is e.g., ...

Than stop polluting this issue. 
Your comment will not change anything and annoy many people which are subscribed to the repository / issue.

**Comment by SomeoneElseOSM** [comment]
(for reference see the related help question at https://help.openstreetmap.org/questions/73116/how-to-check-how-a-tag-renders )

A reply of "stop polluting this issue" is perhaps a little strong, especially as the very first line of the very first entry above was "I think we should show not just shops and amenities, but also some objects with craft=* and office=* key".  I understand that https://github.com/gravitystorm/openstreetmap-carto/issues/1697#issuecomment-580048697 means that essentially it's not really possible to work on this right now, but perhaps the poster does not?

I can fully understand that repeated requests from "customers" (whether it's things like this issue or whether it's me elsewhere banging on about the unsuitability of OSM Carto for hiking routes) can be annoying the umpteenth time you've heard a particular request from different people, but perhaps a more polite response would have been "I'm sorry, but due to the technical limitations we're currently working under - the data in the database that the main OSM.org maps are rendered from - we simply can't fix this at the moment, although work (see the links from https://github.com/gravitystorm/openstreetmap-carto/pull/4015 ) is ongoing".

**Comment by jeisenbe** [comment]
It is now possible to submit a PR to render this feature. Since v5.0.0 all craft=* features will be imported as polygons when mapped as closed ways.

**Comment by TheAdventurer64** [comment]
What about `craft=catering`?

**Comment by kocio-pl** [comment]
There are already multiple craft objects with usage above 2k (see https://taginfo.openstreetmap.org/keys/craft#values ). I guess pink color from the old healthcare objects could work, but nobody has tested it yet. Also the icon codes from gist designed by @Tomasz-W are not accessible now, so we lack files to test them and some new ones should be created (or just reposted).

**Comment by Tomasz-W** [comment]
> Also the icon codes from gist designed by @Tomasz-W are not accessible now, so we lack files to test them and some new ones should be created (or just reposted).

In case they will be needed, I can reupload it. Just contact me ;)

**Comment by kocio-pl** [comment]
Is there anybody willing to try to prepare the code?

**Comment by jeisenbe** [comment]
We still lack consensus on #3635 and it is not at all certain that there will be consensus to add specific icons for certain `craft=*` features.

**Comment by Adamant36** [comment]
> We still lack consensus on #3635 and it is not at all certain that there will be consensus to add specific icons for certain craft=* features.

There should really be a time limit or something on it or something. Or at least intermediate alternative/options in the meantime. Heck even some kind of specific "review rendering X category of icons" issues would be better then nothing. Otherwise, there will be no incentive or will for it to be resolved. Things with it aren't going to just magically improve or be dealt with by referencing the issue every time someone wants to add a new icon either. Ultimately, rendering new things and dropping rendering of older ones should just be a normal part of style maintenance IMO. Obviously priorities will change over time. There's zero reason it should be a (big) thing and it's not fair to anyone involved, or OSM more generally, that it is.  

Even more (or less? I'm not sure) so though because there doesn't seem to be the will to alter the "style purpose" file or whatever it is and it still has a mapper feedback as part of that purpose. I know in this case at least that rendering craft icons help a lot in that regard. At least in my experience there seems to be a lot of miss-tagging of craft places as shops or other things that do render.

**Comment by imagico** [comment]
As @jeisenbe wrote mentioning #3635 has the purpose of making it clear to developers that while some maintainers might express encouragement to develop a change to add new symbols to this style there is no consensus on that among maintainers and such changes are therefore currently likely to be rejected.

#3635 will not go away by letting time pass.  It will be resolved by either all maintainers showing a willingness to develop a consensus for the strategic direction of the style in that regard or by developers developing changes that resolve the issues pointed out in #3635 and that achieve consensus among maintainers practically.

The discussion in #3635 is open and anyone willing to work towards a consensus there is welcome.  @jeisenbe and me also have in #3951 worked towards ideas how to ensure consensus based decision making works in cases some maintainers express a sustained non-constructive attitude.

**Comment by Adamant36** [comment]
> As @jeisenbe wrote mentioning #3635 has the purpose of making it clear to developers that while some maintainers might express encouragement to develop a change to add new symbols to this style there is no consensus on that among maintainers and such changes are therefore currently likely to be rejected.

I know. I just don't think pointing to an issue that hasn't had a comment on it for a year except for ones that are off topic every time someone wants to implement something new really moves things forward at all. Even #3951 hasn't had any activity since January. There is a point where things will just natural proceed how they were before the issues were opened if no one takes the time to actually resolve them. Like I said in my other comment anyway, it's not a black and white thing where they either get dealt with or they don't. There's intermediate steps that can taken in the meantime. A few have been suggested, but from what I remember you tossed them off as not adequate because they didn't completely reinvent the wheel or whatever (I.E. weren't system wide changes), but you can only eat the elephant one part at a time. Books are written a page at a time. The journey of a thousands starts with...Etc etc..Otherwise, it can be akin to analysis paralysis. 

You say that it can be resolved by "developers developing changes that resolve the issues pointed out in #3635", but from what I remember there wasn't even clear changes pointed out that would resolve it. Let alone any with consensus. There was a lot of generalities and vague comments though. I have some ideas, but they are not "system wide" and are likely to be rejected. Some similar suggestions already have been. No one is going to develop things just because either. Especially since this is a maintainer problem, not a developer one. It's on the maintainers to have a clear direction for the style and make it explicit what's acceptable or not. Not to just be like "do a PR for whatever, and maybe it will accepted and maybe it won't." That's not a good way to do things. At least @kocio-pl is 100% clear in position and what direction he thinks the style should go in. I think @jeisenbe is to some degree also. Although less so. You seem to be less specific in what exactly you want done and more so about what you don't. That's just my reading of it though. I could be wrong, but I've read through all the discussions a bunch of times.

**Comment by imagico** [comment]
@Adamant36 - If you want to discuss our decision making processes here then #3951 is the place to do that - not a specific design issue like this.  Same for the general strategy on POI rendering - that is a matter for #3635, not here.

**Comment by kocio-pl** [comment]
If you think they are off-topic, simply don't introduce them.

**Comment by imagico** [comment]
@kocio-pl - sorry, i have no idea what you mean.

**Comment by kocio-pl** [comment]
> As @jeisenbe wrote mentioning #3635 has the purpose (...)

It is you trying to discuss some other issue here.

**Comment by imagico** [comment]
I was explaining to @Adamant36 in more detail the relevance of mentioning #3635 on this issue and encouraged further discussion on generic matters beyond the scope of this issue on the relevant other issues.  I repeated this suggestion more strongly today after further generic commentary without specific connection to this issue.  Your comments do not seem to make much sense in that context to me.  If you want to discuss the way we structure communication on this issue tracker into separate issues that would deserve possibly opening a new issue.

**Comment by kocio-pl** [comment]
But you have not discussed it there, but here, yourself in the first place. Why did you not tell to @jeisenbe to move there and even joined him, but @Adamant36 when he replied to you on your own topic?

**Comment by imagico** [comment]
Sorry @kocio-pl - i can't follow your logic here.  In any case - if you want to discuss procedure or communication style please do so at a separate issue.

**Comment by kocio-pl** [comment]
Do not treat the same topic from different persons in a different way, as you did here.

**Comment by map-per** [comment]
I opend a new PR for craft (#4809) and would be happy to get some input about the design choices: Do you like the colour choice? Or should craft be displayed in office blue? And should rendering start at 17, like for shops, or rather at 18, like for offices?

**Comment by imagico** [comment]
We traditionally do not tie the colors of POI symbols to OSM keys - see for example [amenity=*](https://imagico.de/map/styleinfo/#style=osmcarto&section=tags&key=amenity) - which are shown in very different colors.

Also to keep in mind that a symbol should be intuitive for what it is meant to depict independent of a-priory knowledge of the meaning of the color used.  To give an example: We would not want to visualize the office of an architect with a building symbol and trust the symbol color to be read as this symbol depicting a service related to the construction of a building rather than a building itself.

**Comment by dch0ph** [comment]
There seems to be too much diversity in `craft` usage to have a "craft colour". Even with a colour, we're still missing a strategy for how to deal with craft. 

It might be useful to consider how the map would look with improved handling of craft.

For industrial areas, these are often broken down in `man_made=works`. This generates a label, and this seems fine - there's no demand for rendering `product`. I was going to raise a separate issue about improving `man_made=works` rendering, e.g. moving away from a landuse-style label (overlapping with the labelling of an overall `landuse=industrial`) and using a POI symbol, e.g. generic factory symbol.

The opposite end is `landuse=retail`, where we have detailed POI symbols for `shop=X`. There are tensions over the number of symbols (#3635), but the overall concept is working. 

In terms of `landuse=commercial`, `office=X` is rendered with dot plus name, which seems to be working (there has been no real demand for creating specific office POI symbols), even if we would like to remove the "catch all", e.g. by only rendering `office=X` if `X` appears e.g. more than 10 times in the database.

But the other aspect of `landuse=commercial`, small traders / businesses, typically covered by `craft` is a problem, and we have lots of unrendered features in "trading estates". Some of these will be public-facing (with "shop" character) and some will be more like offices.

TL;DR I do think it's worth exploring whether we can handle most craft features within our current symbology / colour palette. For example `craft=electronics_repair` is not really different to `shop=car_repair`.

**Comment by imagico** [comment]
> For example craft=electronics_repair is not really different to shop=car_repair.

Indeed.  But that can't be generalized to all `craft=*`.  Some are like this - the tag essentially implies a business open to walk-in customers.  But not all of them.  Looking at some of the most commonly tagged `craft=*` for example:

* `craft=winery` - used both for places open for direct sale as well as pure production facilities.  Also used both for _craft wineries_ and industrial production plants equally (in contrast to `craft=brewery` and `industrial=brewery` - where some differentiation takes place though the delineation is not quite clear).
* `craft=sawmill` is a competing tag/synonym for `industrial=sawmill` with no semantic difference being even claimed.

This needs to be looked at on a case-by-case basis - both in terms of suitability for rendering here and in terms of symbol color to use.

**Comment by dieterdreist** [comment]
despite the usage (likely pushed by some editor presets) according to this map: https://taginfo.openstreetmap.org/tags/shop=alcohol#map  "alcohol shops" are not a common shop typology in many/some countries, e.g. in Italy, France or Germany, rather you can buy alcohol "everywhere" (like kiosks, supermarkets, convenience stores, gas stations, etc.) and dedicated shops are usually specialized (e.g. wine shops, or whiskey shops). Craft=winery should NOT get amended with shop=alcohol but with shop=wine if the also sell directly to end customers. Btw., an interesting distinction/attribute would be wine shops that sell "unpackaged" wine, i.e. you bring your own container.
